### ISSUE TITLE
Tab title is reset for a moment when navigating to a dynamic page

### ISSUE THREAD

**Issue by A1yks** [vercel/next.js#65872]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/focused-moser-hh9nk8

### To Reproduce

1. Start the application usign `npm run dev`
2. Open sandbox preview in a new browser tab to be able to see the tab title
3. Click on a link that takes you to another page while looking at the tab title
4. If you did not notice the title reset, then refresh the page and do step 3 again

https://github.com/vercel/next.js/assets/35101788/9e53be20-e9e2-42d7-915c-69dd56384e61

### Current vs. Expected behavior

Tab title shouldn't reset when navigating to another page

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Sun Aug  6 20:05:33 UTC 2023
  Available memory (MB): 4102
  Available CPU cores: 2
Binaries:
  Node: 20.9.0
  npm: 9.8.1
  Yarn: 1.22.19
  pnpm: 8.10.2
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: 14.2.1
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Navigation

### Which stage(s) are affected? (Select all that apply)

next dev (local), next start (local)

### Additional context

I noticed that it doesn't happen when there is no loading.tsx file

**Comment by icyJoseph** [comment]
> I noticed that it doesn't happen when there is no loading.tsx file

This was gonna be my first question. 

More easily visible with watch expressions: 

https://github.com/vercel/next.js/assets/21013447/79692bc0-d865-42f2-8d3e-d6e25dd1938b

`document.title` becomes `''`

I know of another issue that also goes away when there is no loading.tsx... 🤔

**Comment by Avik-creator** [comment]
I guess since you are using redirect.replace(), it might lead to changes in the DOM

**Comment by A1yks** [comment]
> I guess since you are using redirect.replace(), it might lead to changes in the DOM.

Why should it?

**Comment by Avik-creator** [comment]
@A1yks I am sorry for providing the wrong info, but as mentioned in the docs of Next.js:  

https://nextjs.org/docs/app/building-your-application/optimizing/metadata

<img width="655" alt="Screenshot 2024-05-22 at 10 17 00 PM" src="https://github.com/vercel/next.js/assets/77090462/c20024dc-4aeb-49f5-89dc-98e850a63b6d">


More specifically you see the section mentioned above in the screenshot.

**Comment by A1yks** [comment]
@Avik-creator Yes, it is expected for page.js to replace the title from layout.js. The problem is while the page.js is loading, the title becomes empty for a while, as you could see from the video I attached
